alien4cloud/alien4cloud-go-client
# Alien4Cloud go client

Added a go client for Alien4Cloud REST API.
Added example showing its usage to :
* create and deploy an application from a template in Alien4Cloud catalog
* run a workflow
* undeploy and delete the application

## How to verify

Following READMEs provided under:
* examples/create-deploy-app/
* examples/run-workflow/
* examples/undeploy-delete-app/

first upload the Ystia Forge welcome application in the Alien4Cloud catalog,
then build the first example:

```bash
cd examples/create-deploy-app
go build -o deploy
```

Run this example:
```bash
./deploy -url https://1.2.3.4:8088 \
         -user myuser \
         -password mypasswd \
         -app myapp \
         -template org.ystia.samples.topologies.welcome_basic
```

(here the first suited location is used, you can specify a location using argument `-location` if you prefer).

Check you see logs of the on-going deployment.
Once the deployement is done, check the URL of the deployed web app is printed, and the example exits without error.
Check you can access this Web app URL.

Build the workflow example:

```bash
cd examples/run-workflow
go build -o run
```

Run a workflow stopping the web server :

```bash
./run -url https://1.2.3.4:8088 \
      -user myuser \
      -password mypasswd \
      -app myapp \
      -worflow stopWebServer
```

Check you see logs of the on-going workflow. Once done, check you can't access the Web App URL.

Run a workflow starting the web server :

```bash
./run -url https://1.2.3.4:8088 \
      -user myuser \
      -password mypasswd \
      -app myapp \
      -worflow startWebServer
```

 Check you see logs of the on-going workflow. Once done, check you can access the Web App URL.
 
Build the undeploy example:


```bash
cd examples/undeploy-delete-app
go build -o undeploy
```

undeploy and delete the application:

```bash
./undeploy -url https://1.2.3.4:8088 \
           -user myuser \
           -password mypasswd \
           -app myapp
```

Check undeployment logs appear.
Once done, login on Alien4Cloud UI and confirm the application has been deleted.
